docs/docs/view-config-js.md

```diff
@@ -0,0 +1,474 @@
+---
+id: view-config-js
+title: View Configs via JS API
+sidebar_label: View Configs via JS API
+slug: /view-config-js
+---
+
+## Overview
+
+The Vitessce view config defines how data is retrieved, which visualization components are rendered, and how different components are coordinated. Ultimately, this configuration must be a JSON object when it is passed to the `<Vitessce/>` React component's `config` prop. Writing large JSON objects by hand can be difficult, so we have developed object-oriented APIs to simplify this process. There are corresponding APIs in [Python](https://vitessce.github.io/vitessce-python/) and [R](https://vitessce.github.io/vitessce-r/) if one of those languages is more familiar to you.
+
+
+## `VitessceConfig`
+
+`VitessceConfig` is a class representing a Vitessce view config. To begin creating a view config with the API, you will need to instantiate a `VitessceConfig` object.
+The methods of this object (and the objects its methods return) allow you to manipulate the underlying configuration.
+When you are ready, you will use the `.toJSON()` method to convert the `VitessceConfig` object to a plain JSON object.
+
+
+### `constructor(name, description)`
+
+Construct a Vitessce view config object.
+
+
+#### Parameters:
+- `name` (`string`) - A name for the view config.
+- `description` (`string`) - A description for the view config. Optional.
+
+```js {3}
+import { VitessceConfig } from 'vitessce';
+
+const vc = new VitessceConfig("My config");
+```
+
+
+### `addDataset(name)`
+
+Add a dataset to the config.
+
+#### Parameters:
+- `name` (`string`) - A name for the dataset.
+
+#### Returns:
+- Type: `VitessceConfigDataset`
+
+Returns the instance for the new dataset.
+
+
+```js {4}
+import { VitessceConfig } from 'vitessce';
+
+const vc = new VitessceConfig("My config");
+const dataset = vc.addDataset("My dataset")
+    .addFile(
+        "http://example.com/my-cells-data.json",
+        "cells",
+        "cells.json"
+    );
+```
+
+
+### `addView(dataset, component, extra)`
+
+Add a view to the config.
+
+#### Parameters:
+- `dataset` (`VitessceConfigDataset`) - A dataset instance to be used for the data visualized in this view.
+- `component` (`string`) - A component name.
```

Comment:
I think we should link back to the list of components here (and anywhere else there are "magic strings" that are actually constants.)